src/login/pam_systemd.c

```diff
@@ -945,8 +945,10 @@ static void session_context_mangle(
                  * probably because it wants to set it to something as pam_time/pam_access/… require PAM_TTY to be set
                  * (as they otherwise even try to update it!) — but cron doesn't actually allocate a TTY for its forked
                  * off processes.) */
-                c->type = "unspecified";
-                c->class = "background";
+                if (isempty(c->type))
+                        c->type = "unspecified";
+                if (isempty(c->class))
+                        c->class = "background";
```

Comment:
lgtm, but can you extend the comment, and say explicitly that we want to allow users to configure background-light here if they want, and hence only fill this in if not specified otherwise.

Reply:
also, can you do something similar for the ssh case below, i.e. fill in class only if not set as well. so that people have the chance to set the ssh session class to "user-light" if they want.

Reply:
and maybe also in the "systemd-user" case above, so that people can chose between "manager-early" and "manager" if they like, just by dropping a dropin file in for user@.service for some user that sets XDG_SESSION_CLASS=… env var.